### PR TITLE
[api] fix runservice api docu via token

### LINKED ISSUE #10395: API documentation mentions non-existent command "createtoken"
The API documentation mentions a non-existent command "createtoken" for the `POST /trigger/runservice` route:
https://github.com/openSUSE/open-build-service/blob/c5bfd58798df61e780338794f06a981939ca6131/docs/api/api/api.txt#L1436

Should this be `POST /person/<userid>/token` instead?

### PR DESCRIPTION
**Pull request by adrianschroeter**
fixes #10395

### CODE REVIEW

File: docs/api/api/api.txt [hunk 1]
```diff
@@ -1433,14 +1433,16 @@ XmlResult: buildinfo
 
 POST /trigger/runservice
 
-  This call needs a token created via "createtoken" command to identify user and
-  package which shall be updated via a source service. The token must be delivered
-  as HTTP header:
+  This call needs a token to identify user and package which shall be updated
+  via a source service. See the cmd=create in the /person/<userid>/token route.
+
+  The token itself must be delivered as HTTP header:
```
Comment (dcermak) [inline docs/api/api/api.txt:1439]:
```suggestion
  The token itself must be delivered as a HTTP header:
```

File: docs/api/api/api.txt [hunk 2]
```diff
@@ -1433,14 +1433,16 @@ XmlResult: buildinfo
 
 POST /trigger/runservice
 
-  This call needs a token created via "createtoken" command to identify user and
-  package which shall be updated via a source service. The token must be delivered
-  as HTTP header:
+  This call needs a token to identify user and package which shall be updated
```
Comment (dcermak) [inline docs/api/api/api.txt:1436]:
```suggestion
  This call needs a token to identify the user and package which shall be updated
```

File: docs/api/api/api.txt [hunk 3]
```diff
@@ -1433,14 +1433,16 @@ XmlResult: buildinfo
...
+
+  The token itself must be delivered as HTTP header:
     Authorization: Token <TOKEN_STRING>
 
 Parameters
-  project: To be used together with "package" to identify the object.
-  package:
+  project: To be used together with "package" to identify the object. Must not be used
+           when the token is bound to a package.
+  package: see project parameter.
```
Comment (dcermak) [inline docs/api/api/api.txt:1445]: What does this mean? That the package parameter must not be used in the same way the project parameter shall not be used when the token is bound to a package?

**Review by dcermak — COMMENTED** [review]
I've added a few suggestions

File: docs/api/api/api.txt (hunk 3 above)
Comment (adrianschroeter) [reply docs/api/api/api.txt:1445]: You always need project and package parameter together. Or neither of them.

Comment (adrianschroeter) [reply docs/api/api/api.txt:1445]: (in future the project parameter alone may work though, but this is not implemented)

Comment (dcermak) [reply docs/api/api/api.txt:1445]: > You always need project and package parameter together. Or neither of them.

Ah, ok. That part was not clear to me

File: docs/api/api/api.txt [hunk 4]
```diff
@@ -1433,14 +1433,16 @@ XmlResult: buildinfo
...
+  This call needs a token to identify the user and package which shall be updated
+  via a source service. See the cmd=create in the /person/<userid>/token route.
+
+  The token itself must be delivered as a HTTP header:
     Authorization: Token <TOKEN_STRING>
 
 Parameters
-  project: To be used together with "package" to identify the object.
-  package:
+  project: To be used together with "package" to identify the object. Must not be used
```
Comment (dcermak) [inline docs/api/api/api.txt:1443]:
```suggestion
  project: Must be used in conjunction with the "package" parameter to identify the service to trigger. Must not be used
```

File: docs/api/api/api.txt [hunk 5]
```diff
@@ -1433,14 +1433,16 @@ XmlResult: buildinfo
...
+           when the token is bound to a package.
```
Comment (dcermak) [inline docs/api/api/api.txt:1444]:
```suggestion
           when the token is bound to a specific package.
```

File: docs/api/api/api.txt (hunk 3 above)
Comment (dcermak) [reply docs/api/api/api.txt:1445]: Just out of interest, what does happen if you pass the project & package parameters but the token is bound to a specific package?

Comment (adrianschroeter) [reply docs/api/api/api.txt:1445]: an error is reported IIRC

File: docs/api/api/api.txt (hunk 4 above)
Comment (adrianschroeter) [reply docs/api/api/api.txt:1443]: hm, it violates our style a bit. One can not check easily the existing parameters here anymore in the list when the "package" parameter is only inside of the text

**Comment by dcermak** [comment]
Anything that blocks this from getting merged?

**Comment by vpereira** [comment]
> Anything that blocks this from getting merged?

I think the comments from @adrianschroeter ?

File: docs/api/api/api.txt (hunk 4 above)
Comment (dcermak) [reply docs/api/api/api.txt:1443]: But the `package:` parameter is still present in a comparable fashion to how it used to be there. I'm just wondering what the issue here is?

**Review by hennevogel — APPROVED** [review]

**Pull request merged**